infra/examples-dev/aws-all/variables.tf

```diff
@@ -0,0 +1,251 @@
+    error_message = "The `environment_name` cannot start with 'aws' or 'ssm', as this will name your AWS resources with prefixes that displease the AMZN overlords."
+  }
+}
+
+variable "aws_account_id" {
+  type        = string
+  description = "id of aws account in which to provision your AWS infra"
+
+  validation {
+    condition     = can(regex("^\\d{12}$", var.aws_account_id))
+    error_message = "The aws_account_id value should be 12-digit numeric string."
+  }
+}
+
+variable "aws_assume_role_arn" {
+  type        = string
+  description = "ARN of role Terraform should assume when provisioning your infra. (can be `null` if your CLI is auth'd as the right user/role)"
+  default     = null
+}
+
+variable "aws_region" {
+  type        = string
+  description = "default region in which to provision your AWS infra"
+  default     = "us-east-1"
+}
+
+variable "aws_ssm_param_root_path" {
+  type        = string
+  description = "root to path under which SSM parameters created by this module will be created; NOTE: shouldn't be necessary to use this is you're following recommended approach of using dedicated AWS account for deployment"
+  default     = ""
+
+  validation {
+    condition     = length(var.aws_ssm_param_root_path) == 0 || length(regexall("/", var.aws_ssm_param_root_path)) == 0 || startswith(var.aws_ssm_param_root_path, "/")
+    error_message = "The aws_ssm_param_root_path value must be fully qualified (begin with `/`) if it contains any `/` characters."
+  }
+}
+
+variable "caller_gcp_service_account_ids" {
+  type        = list(string)
+  description = "ids of GCP service accounts allowed to send requests to the proxy (eg, unique ID of the SA of your Worklytics instance)"
+  default     = []
+
+  validation {
+    condition = alltrue([
+      for i in var.caller_gcp_service_account_ids : (length(regexall("^\\d{21}$", i)) > 0)
+    ])
+    error_message = "The values of caller_gcp_service_account_ids should be 21-digit numeric strings."
+  }
+}
+
+variable "caller_aws_arns" {
+  type        = list(string)
+  description = "ARNs of AWS accounts allowed to send requests to the proxy (eg, arn:aws:iam::914358739851:root)"
+  default     = []
+
+  validation {
+    condition = alltrue([
+      for i in var.caller_aws_arns : (length(regexall("^arn:aws:iam::\\d{12}:((role|user)\\/)?\\w+$", i)) > 0)
+    ])
+    error_message = "The values of caller_aws_arns should be AWS Resource Names, something like 'arn:aws:iam::123123123123:root'."
+  }
+}
+
+
+
+
+variable "salesforce_domain" {
+  type        = string
+  description = "Domain of the Salesforce to connect to (only required if using Salesforce connector). To find your My Domain URL, from Setup, in the Quick Find box, enter My Domain, and then select My Domain"
+  default     = ""
+}
+
+variable "jira_server_url" {
+  type        = string
+  default     = null
+  description = "(Only required if using Jira Server connector) URL of the Jira server (ex: myjiraserver.mycompany.com)"
```

Comment:
rename to `jira_server_host`? looks more appropriate

Reply:
host would imply it doesn't include a path ... which it might, right?

@aperez-worklytics ?  we ideally should be consistent everywhere; jira via proxy not really released yet, so now's the time to make a change